dns/cloudflare.py

```diff
@@ -46,6 +46,8 @@ def request(method, action, param=None, **params):
         params = json.dumps(params)
     else:
         params = urllib.urlencode(params)
+    if method == 'GET':
+        action = action + '?' + params
     conn.request(method, '/client/v4/zones' + action, params,
```

Comment:
Yep that's a bug. I misunderstood third parameter in [HTTPSConnection.request](https://docs.python.org/3/library/http.client.html#http.client.HTTPConnection.request) . Good job!